### PR TITLE
Feature 4/create name registration - 1st step

### PR DESCRIPTION
**Pull request by aufgfua**
## Description

Create the screens for [Name Registration (1st step)](https://www.figma.com/file/eaAfUatNrDWpFPbkCsLMYG/Frontend-contribution---Onboarding-process?node-id=1%3A1633)

[Issue #4 ](https://github.com/TheTributeCommunity/fesbal-frontend/issues/4)

## Changes

- Add new page for registration step for i18n lib
- Create registration form
- Create registration page (in the same structure as user ID registration page) - This should be better if refactored to be a multi-step form, and not a multi-page one. But for the future, I believe.

## Checks

- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

image

[Issue #4 ](https://github.com/TheTributeCommunity/fesbal-frontend/issues/4)

### CODE REVIEW

File: src/components/molecules/UserNameForm.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import useUserNameForm from "../../hooks/useUserNameForm";
+import {useTranslation} from "react-i18next";
+import {namespaces} from "../../i18n/i18n.constants";
+
+const UserNameForm = () => {
+    const {userName, userSurname, onSubmit, validateNameSurname, onNameChange, onSurnameChange} = useUserNameForm();
+    const {t: translation} = useTranslation(namespaces.pages.registrationName);
+
+    const isValidText = (text: string) => {
+        const unwantedText = text.match(/[^A-Za-záàâãéèêíïóôõöúçñÁÀÂÃÉÈÍÏÓÔÕÖÚÇÑ~^´` ]+/ig) || [];
```
Comment (rdiaz82) [inline src/components/molecules/UserNameForm.tsx:11]: I think that we should find for a more curated way of validation

File: src/hooks/useUserNameForm.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import {useNavigate} from "react-router-dom";
+
+const useUserNameForm = () => {
+    const [userName, setUserName] = useState<string>('');
+    const [userSurname, setUserSurname] = useState<string>('');
+    const MIN_LENGTH = 2
+
+    const navigate = useNavigate();
+    const validateNameSurname = (): boolean => {
+        return userName.length >= MIN_LENGTH && userSurname.length >= MIN_LENGTH;
```
Comment (rdiaz82) [inline src/hooks/useUserNameForm.ts:11]: one question, why do we have the validation separated in two different places and validating different things? shouldn't be easier to have all the validations in the same place?

Comment (aufgfua) [reply src/hooks/useUserNameForm.ts:11]: I renamed it for isLettersOnly. My idea here was that:

1 - The inputs should be text only, so I believe this validation should be immediate on typing. That's why I put it on the input's onChange property. This serves only to block unwanted characters.
2 - The name and surname validation refers to any wanted logic around if the name should be accepted for submission.

So one of the validations (isLettersOnly) is used to check if a typed character should be accepted or blocked. It's used on typing. If you try to type **;** (example) it won't appear.
The other (min text length) is used to check if a text should be submitted. It's used for submission and to enable the submit button.

**Review by rdiaz82 — APPROVED** [review]
LGTM

**Pull request merged**